python/export_util.py

```diff
@@ -235,3 +311,287 @@ def csv_query(
+                )
+            output.write("</node>\n")
+
+        elif element.get("type") == "relationship":
+            output.write(
+                f'<edge id="e{str(element.get("id"))}" source="n{str(element.get("start"))}" target="n{str(element.get("end"))}" label="{element.get("label")}">'  # noqa: E501
+            )
+
+            if config.get("format").upper() == "TINKERPOP":
+                output.write(
+                    f'<data key="labelE">{element.get("label")}</data>'
+                )
+            else:
+                output.write(
+                    f'<data key="label">{element.get("label")}</data>'
+                )
+            if config.get("format").upper() == "GEPHI":
+                output.write(f'<data key="TYPE">{element.get("label")}</data>')
+
+            for key, value in element.get("properties").items():
+                output.write(
+                    f'<data key="{key}">{get_value_string(value)}</data>'
+                )
+            output.write("</edge>\n")
+
+
+def write_footer(output: io.StringIO):
+    output.write("</graph>\n")
+    output.write("</graphml>")
+
+
+def set_default_config(config: mgp.Map):
+    if not config.get("stream"):
+        config.update({"stream": False})
+    if not config.get("format"):
+        config.update({"format": ""})  # should it be Gephi?
+    if not config.get("caption"):
+        config.update({"caption": []})
+    if not config.get("useTypes"):
+        config.update({"useTypes": False})
+    if not config.get("leaveOutLabels"):
+        config.update({"leaveOutLabels": False})
+    if not config.get("leaveOutProperties"):
+        config.update({"leaveOutProperties": False})
+    if (
+        not isinstance(config.get("stream"), bool)
+        or not isinstance(config.get("format"), str)
+        or not isinstance(config.get("caption"), list)
+        or not isinstance(config.get("useTypes"), bool)
+        or not isinstance(config.get("leaveOutLabels"), bool)
+        or not isinstance(config.get("leaveOutProperties"), bool)
+    ):
+        raise TypeError(
+            "Config parameter must be a map with specific \
+             keys and values described in documentation."
+        )
+
+
+@mgp.read_proc
+def graphml(
+    ctx: mgp.ProcCtx,
+    path: str = "",
+    config: mgp.Map = {},
+) -> mgp.Record(status=str):
+    """
+    Procedure to export the whole database to a graphML file.
+
+    Parameters
+    ----------
+    path : str
+        Path to the graphML file containing the exported graph database.
+    config : Map
+
+    """
+
+    graph = get_graph(ctx, 1)
+    set_default_config(config)
+
+    if config.get("leaveOutLabels") or config.get("leaveOutProperties"):
+        for element in graph:
+            if config.get("leaveOutLabels") and element.get("type") == "node":
+                element.update({"labels": []})
+            if config.get("leaveOutProperties"):
+                element.update({"properties": {}})
+
```

Comment:
For me it makes sense to pass these config settings to get graph function as part of some config. So for example, get_graph can accept ctx and config dict {graphML:false, leaveOutLabels:false, leaveOutProps:false} can be default and inside that function we do additional processing. I think it is bit faster because geting of properties is expensive (O(n) for each node) but in case of setting leaveOutProperties we are doing that in any case